tableau_utilities/scripts/datasource.py

```diff
@@ -0,0 +1,85 @@
+from pathlib import Path
+
+from tableau_utilities.tableau_file.tableau_file import Datasource
+from tableau_utilities.tableau_file.tableau_file import TableauFile
+from tableau_utilities.tableau_file.tableau_file_objects import Column
+from tableau_utilities.scripts.server_operate import fill_in_id_name_project, get_object_list
+
+
+def save_datasource(server, datasource_id, datasource_name, project_name, include_extract, debugging_logs):
+    """ Gets the datasource_id if needed and downloads the datasource
+
+    Args:
+        datasource_id: The datasource id
+        datasource_name: The name of the datasource
+        project_name: The name of the project the datasource is in
+        include_extract: True to include the extract. Downloading may take a long time
+        debugging_logs: Print information for debugging to the console
+
+
+    """
+
+    object_list = get_object_list(object_type='datasource', server=server)
+    datasource_id, datasource_name, project_name = fill_in_id_name_project(datasource_id, datasource_name,
+                                                                           project_name, object_list)
+    if debugging_logs:
+        print(
+            f'GETTING DATASOURCE ID: {datasource_id}, NAME: {datasource_name}, PROJECT NAME: {project_name}, INCLUDE EXTRACT {include_extract}')
+
+    datasource_path = server.download_datasource(datasource_id, include_extract)
+
+    if debugging_logs:
+        print(
+            f'DATASOURCE PATH: {datasource_path}')
+
+    return datasource_path
+
+
+def save_xml(datasource_path):
+    """ Extracts XML from a tdsx file and saves as a raw .xml file
+
+    """
+
+    xml_path = TableauFile(datasource_path).save(save_raw_xml=True)
+    print(f'XML SAVED TO: {xml_path}')
+
+
+def datasource(args, server=None):
+
+    debugging_logs = args.debugging_logs
+
+    datasource_source = args.datasource_source
+    datasource_path = args.datasource_path
+    datasource_id = args.datasource_id
+    datasource_name = args.datasource_name
+    project_name = args.datasource_project_name
+    include_extract = args.include_extract
+
+    if datasource_source == 'online':
+        datasource_path = save_datasource(server, datasource_id, datasource_name, project_name, include_extract, debugging_logs)
+
+    if args.datasource_save_xml:
+        save_xml(datasource_path)
+
+    if datasource_path is not None:
+        datasource = Datasource(datasource_path)
+
+    if args.column:
```

Comment:
@JustinGrilli  I can't seem to get this column stuff to work and it's the last part for having all functionality in the CLI.

Do you have time this week to pair on this?  Perhaps we could get it working together and then discuss any other changes you want to this for me to get it into mergeable shape.

Reply:
Sure if you want to schedule some time after 4pm PT, any day this week, that would work for me!